src/FinfoMimeTypeDetector.php

```diff
@@ -39,23 +39,38 @@ class FinfoMimeTypeDetector implements MimeTypeDetector, ExtensionLookup
      */
     private $inconclusiveMimetypes;
 
+    /**
+     * @var bool
+     */
+    private $ignoreNonSeekableStreams;
+
+    /**
+     * Buffer size to read from streams if no other bufferSampleSize is defined.
+     */
+    const STREAM_BUFFER_SAMPLE_SIZE_DEFAULT = 4100;
+
     public function __construct(
         string $magicFile = '',
         ExtensionToMimeTypeMap $extensionMap = null,
         ?int $bufferSampleSize = null,
-        array $inconclusiveMimetypes = self::INCONCLUSIVE_MIME_TYPES
+        array $inconclusiveMimetypes = self::INCONCLUSIVE_MIME_TYPES,
+        bool $ignoreNonSeekableStreams = true
     ) {
         $this->finfo = new finfo(FILEINFO_MIME_TYPE, $magicFile);
         $this->extensionMap = $extensionMap ?: new GeneratedExtensionToMimeTypeMap();
         $this->bufferSampleSize = $bufferSampleSize;
         $this->inconclusiveMimetypes = $inconclusiveMimetypes;
+        $this->ignoreNonSeekableStreams = $ignoreNonSeekableStreams;
```

Comment:
with the other suggestion implemented, this configuration option should be removed

Reply:
Done. Tests adjusted accordingly.
Currently unhappy about using `http://` as a non-seekable stream but couldn't come up with a better idea as of yet.
Do you by chance know if there's some phpunit stream mocks that could do that?